ci/pinned: update

wolfgangwalther opened the pull request:
This gives us Nix 2.30 and nixfmt 1.0.0.

From the nixpkgs-unstable channel:
https://hydra.nixos.org/eval/1817034#tabs-inputs

Changes for treefmt-nix:
https://github.com/numtide/treefmt-nix/compare/a05be418a1af1198ca0f63facb13c985db4cb3c5...421b56313c65a0815a52b424777f55acf0b56ddf

## Things done

- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wolfgangwalther committed:
ci/pinned: update

This gives us Nix 2.30 and nixfmt 1.0.0.

From the nixpkgs-unstable channel:
https://hydra.nixos.org/eval/1817034#tabs-inputs

Changes for treefmt-nix:
https://github.com/numtide/treefmt-nix/compare/a05be418a1af1198ca0f63facb13c985db4cb3c5...421b56313c65a0815a52b424777f55acf0b56ddf

wolfgangwalther committed:
treewide: run nixfmt 1.0.0

MattSturgeon commented:
Isn't there a nix 2.3.1 with the CVE fix? Can't we just bump the nix_2_30 package and block the CI pin until that is available?

Also, due to changes in nixfmt we will need to do a treewide reformat and add that commit to git-ignore-revs.

wolfgangwalther committed:
nixos/doc/wireless: fix comments

Those had been misplaced by nixfmt.

wolfgangwalther committed:
treewide: run treefmt with mdcr/nixfmt

infinisil commented:
@wolfgangwalther We're having the Nix formatting team meeting right now, want to join to discuss this? https://meet.zrh.init7.net/nix-formatting

jfly reviewed (approved):
We (the Nix Formatting Team) have reviewed the nixfmt related changes, which LGTU.

wolfgangwalther committed:
treewide: fix syntax errors in nix code blocks

Fixes all code blocks with "nix" language in markdown files for syntax
errors to be able to run nixfmt in the next step.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/formatting-team-meeting-2025-07-22/67066/1

Ma27 commented:
I guess the reformat should also be added to `.git-blame-ignore-revs`?

wolfgangwalther committed:
ci/parse: raise minimum test to Nix 2.24

Nix 2.3 is marked as insecure and thus not cached anymore.

We'll either need to patch it and cache it again or drop it.
